flatland_plugins/src/laser.cpp

```diff
@@ -230,6 +241,10 @@ void Laser::ParseParameters(const YAML::Node &config) {
                         boost::algorithm::join(invalid_layers, ",") + "}");
   }
 
+  std::vector<std::string> reflectance_layer = {"reflectance"};
```

Comment:
Can you comment on special casing the layer named "reflectance"? I'm not a big fan of layer names having special meaning.

It might be more elegant to have the laser scan plugin take a list of layers that trigger reflectance and generate reflectance_layer_bits_ from that (that way any zero or more layers can generate intensity data for any laser if desired). layers_bits_ could then be generated using the same code as layer_bits_, which would be more consistent.

Another question is: 
is there a use case for having a laser that doesn't trigger intensity on a layer that it sees? If not, we can just make intensity: true|false a setting on the laser plugin. If true, any "hits" would return intensity values, and if false they wouldn't. This would result in simpler code and be easier to use.

Reply:
Hi Joseph,

this was a quick hack just to get into the Flatland simulator, so some of the design decisions were taken with that in mind.

As you say, maybe having an special layer it's not the best. It is nicer and cleaner to have the reflectance layers as a parameter list, but that's why I named the variable reflectance_layer**s**_bits_

No, I cannot think right now of a use case where the laser does not return intensity (or it is 0) when hits really something and returns a distance to it. In that case, don't add the footprint of the part to the "reflectance" list of layers, so intensity will be zero (as it is coded now).

What will be more interesting is to implement an intensity decay with the distance. Up to ~ one meter, intensity increases, and then decays proportional to the squared distance.